mypy/messages.py

```diff
@@ -898,8 +898,12 @@ def type_arguments_not_allowed(self, context: Context) -> None:
         self.fail('Parameterized generics cannot be used with class or instance checks', context)
 
     def disallowed_any_type(self, typ: Type, context: Context) -> None:
-        self.fail('Expressions of type "Any" are disallowed '
-                  '(has type {})'.format(self.format(typ)), context)
+        if isinstance(typ, AnyType):
+            infix = 'type'
+        else:
+            infix = 'type that contains type'
+        self.fail('Expressions of {} "Any" are disallowed '
+                  '(has type {})'.format(infix, self.format(typ)), context)
```

Comment:
Also `Expressions of type "Any" are disallowed (has type "Any")` is redundant, which could be confusing.